lib/core/src/display/rendering/htmlobject.rs

```diff
@@ -0,0 +1,76 @@
+use super::Object;
+use crate::system::web::create_element;
+use crate::system::web::dyn_into;
+use crate::system::web::Result;
+use crate::system::web::Error;
+use crate::system::web::StyleSetter;
+use nalgebra::Vector2;
+use web_sys::HtmlElement;
+
+
+/// A structure for representing a 3D HTMLElement in a `HTMLScene`
+#[derive(Shrinkwrap)]
+#[shrinkwrap(mutable)]
+pub struct HTMLObject {
+    #[shrinkwrap(main_field)]
+    pub object : Object,
+    pub element : HtmlElement,
+    pub dimensions : Vector2<f32>,
+}
+
+impl HTMLObject {
+    /// Creates a HTMLObject from element name
+    /// # Example
+    /// ```rust,no_run
+    /// use basegl::display::rendering::HTMLObject;
+    /// let object = HTMLObject::new("div");
+    /// ```
+    pub fn new(name: &str) -> Result<Self> {
+        let element = dyn_into(create_element(name)?)?;
+        Ok(Self::from_element(element))
+    }
+
+    /// Creates a HTMLObject from a web_sys::HtmlElement
+    pub fn from_element(element: HtmlElement) -> Self {
+        element.set_property_or_panic("transform-style", "preserve-3d");
+        element.set_property_or_panic("position", "absolute");
+        element.set_property_or_panic("width", "0px");
+        element.set_property_or_panic("height", "0px");
+
+        Self {
+            object: Default::default(),
+            element,
+            dimensions: Vector2::new(0.0, 0.0)
+        }
```

Comment:
For me, the @notdanilo version is better - i looks nearly exactly the same, and I immediately see that this is filling fields of return value. When I see `let` I never know where it can be used

Reply:
Default isn't implemented for Vectors. We need to use Vector2::new(x,y).

I rather use 1-liners. So @wdanilo version looks cleaner for me.
If we use `let` right before `Self`, it it's immediately clear where it's being used.

Reply:
We can write these expressions like this so it be used to keep 1-liners smaller than 80 chars.